src/chains/ethereum/ethereum/src/api.ts

```diff
@@ -1046,6 +1046,99 @@ export default class EthereumApi implements Api {
+   *   without code will return
+   *   `"0xc5d2460186f7233c927e7db2dcc703c0e500b653ca82273b7bfad8045d85a470"`
+   * * `nonce`: `QUANTITY` - the nonce of the account.
+   * * `storageHash`: `DATA` - 32 Bytes - SHA3 of the StorageRoot. All storage
+   *   will deliver a MerkleProof starting with this rootHash.
+   * * `accountProof`: `Array` - Array of rlp-serialized MerkleTree-Nodes,
+   *   starting with the stateRoot-NODE, following the path of the SHA3
+   *   (address) as key.
+   * * `storageProof`: `Array` - Array of storage entries as requested. Each
+   *   entry is an object with the following properties:
+   *   * `key`: `DATA` - the requested storage key.
+   *   * `value`: `QUANTITY` - the storage value.
+   *   * `proof`: `Array` - Array of rlp-serialized MerkleTree-Nodes, starting
+   *     with the storageHash-Node, following the path of the SHA3 (key) as
+   *     path.
+   * @example
+   * ```javascript
+   * // Simple.sol
+   * // // SPDX-License-Identifier: MIT
+   * //  pragma solidity ^0.7.4;
+   * //
+   * //  contract Simple {
+   * //      uint256 public value;
+   * //      constructor() payable {
+   * //          value = 5;
+   * //      }
+   * //  }
+   * const simpleSol = "0x6080604052600560008190555060858060196000396000f3fe6080604052348015600f57600080fd5b506004361060285760003560e01c80633fa4f24514602d575b600080fd5b60336049565b6040518082815260200191505060405180910390f35b6000548156fea26469706673582212200897f7766689bf7a145227297912838b19bcad29039258a293be78e3bf58e20264736f6c63430007040033";
+   * const [from] = await provider.request({ method: "eth_accounts", params: [] });
+   * await provider.request({ method: "eth_subscribe", params: ["newHeads"] });
+   * const txHash = await provider.request({ method: "eth_sendTransaction", params: [{ from, gas: "0x5b8d80", data: simpleSol }] });
+   * await provider.once("message"); // Note: `await provider.once` is non-standard
+   * const txReceipt = await provider.request({ method: "eth_getTransactionReceipt", params: [txHash] });
+   * const proof = await provider.request({ method: "eth_getProof", params: [txReceipt.contractAddress, ["0x0", "0x1"], "latest"] });
+   * console.log(proof);
+   * ```
+   */
+  @assertArgLength(3)
+  async eth_getProof(
+    address: DATA,
+    storageKeys: DATA[],
+    blockNumber: QUANTITY | Ethereum.Tag
+  ): Promise<Ethereum.AccountProof<"private">> {
+    if (this.#blockchain.fallback) {
+      throw new Error(
+        "eth_getProof is not supported on a forked network. See https://github.com/trufflesuite/ganache/issues/3234 for details."
+      );
+    }
+    const blockchain = this.#blockchain;
+    const targetBlock = await blockchain.blocks.get(blockNumber);
+
+    const ganacheAddress = Address.from(address);
+    const bufferAddress = ganacheAddress.toBuffer();
+    const ethereumJsAddress = { buf: bufferAddress, equals: (a:{buf:Buffer}) => a.buf.equals(bufferAddress)} as any;
```

Comment:
I don't think the`getProof` method will ever use the `equals`, so we don't need to create it (code coverage doesn't hit this function).

```suggestion
    const ethereumJsAddress = { buf: bufferAddress } as any;
```

Also, it looks like these files needs to be prettiered.